packages/components/icons/src/inline-svg.tsx

```diff
@@ -0,0 +1,25 @@
+import type { Props } from './templates/icon.styles';
+
+import React from 'react';
+import DOMPurify from 'dompurify';
+import SVG from 'react-inlinesvg';
+import { useTheme } from '@emotion/react';
+import { getIconStyles } from './templates/icon.styles';
+
+type InlineSvgProps = Props & {
+  data: string;
+};
+
+const InlineSvg = (props: InlineSvgProps) => {
+  const theme = useTheme();
+  const sanitized = DOMPurify.sanitize(props.data, {
+    USE_PROFILES: { svg: true },
+    FORBID_ATTR: [
+      // To avoid injection by using `style="filter:url(\"data:image/svg+xml,<svg`
+      'style',
+    ],
+  });
+  return <SVG src={sanitized} css={getIconStyles(props, theme)} />;
```

Comment:
We render the SVG using `react-inlinesvg`, which nicely parses the string into react/dom elements.

It also works with remote URLs, in case one is provided.

<img width="697" alt="image" src="https://user-images.githubusercontent.com/1110551/125061905-1b301780-e0ae-11eb-8f6f-fdb77cdeee54.png">